xbmc/guilib/GUIControl.h

```diff
@@ -219,6 +219,7 @@ class CGUIControl
   virtual void SetInvalid() { m_bInvalidated = true; };
   virtual void SetPulseOnSelect(bool pulse) { m_pulseOnSelect = pulse; };
   virtual std::string GetDescription() const { return ""; };
+  virtual std::string GetDescriptionByIndex(int index) const { return ""; };
```

Comment:
You said:

> there is no need to call super. GetDescription() should be enough.

So I tried that and got:

```
GUIEditControl.cpp: In member function ‘virtual std::string CGUIEditControl::GetDescription(int) const’:                                                          │
GUIEditControl.cpp:754:27: error: no matching function for call to
       return GetDescription();                                                                                                                                     
                             ^                                                                                                                                      
 GUIEditControl.cpp:754:27: note: candidate is:                                                                                                                    
 GUIEditControl.cpp:751:13: note: virtual std::string CGUIEditControl::GetDescription(int) const                                                                   
  std::string CGUIEditControl::GetDescription(int index) const                                                                                                     
              ^                                                                                                                                                    
 GUIEditControl.cpp:751:13: note:   candidate expects 1 argument, 0 provided
```

And since @mkortstiege said:

> Looks good. I don´t really like the duplication of GetDescription though.

Changing the method name solved both issues.